src/transformers/models/paligemma/modeling_paligemma.py

```diff
@@ -282,6 +283,47 @@ def resize_token_embeddings(self, new_num_tokens: Optional[int] = None, pad_to_m
         self.vocab_size = model_embeds.num_embeddings
         return model_embeds
 
+    def construct_causal_mask_with_block_attention(self, attention_mask, input_ids, text_mask, inputs_embeds):
+        # Modified from gemma
+        target_length = attention_mask.shape[-1]
+        dtype, device = inputs_embeds.dtype, inputs_embeds.device
+        min_dtype = torch.finfo(dtype).min
+        sequence_length = input_ids.shape[1]
+        causal_mask = torch.full(
+            (sequence_length, target_length),
+            fill_value=min_dtype,
+            dtype=dtype,
+            device=device,
+        )
+        if sequence_length != 1:
+            causal_mask = torch.triu(causal_mask, diagonal=1)
+
+        mask = input_ids == self.config.prefix_suffix_separator_index
```

Comment:
This is not very robust unfortunately: one or several `\n` tokens might be present in the prefix. Could this depend on the labels passed instead?